File: Aman_Bhaiya_DSA_List/Arrays/LQ34.py
import math


def check_bit(arr,idx):
    return arr[idx >> 5] & (1 << (idx & 31))

def set_bit(arr,idx):
    arr[idx >> 5] |= (1 << (idx & 31))

a = 5
b = 34

arr_size = math.ceil(abs(b-a)/32)
arr = [0] * arr_size

for i in range(a,b+1):
    if i%2 == 0 or i%5 == 0:
        set_bit(arr,i-a)

for i in range(a,b+1):
    if check_bit(arr,i-a):
        print(i)

File: Aman_Bhaiya_DSA_List/Arrays/LQ34.java
package Aman_Bhaiya_DSA_List.Arrays;

public class LQ34 {
    public static Boolean checkBit(int[] arr, int n){
        int bit_value = arr[n >> 5] & (1 << (n & 31));
        return bit_value != 0;
    }

    public static void setBit(int[] arr, int n){
        arr[n >> 5] |= (1 << (n & 31));
    }

    public static void main(String args[]){
        int a = 5;
        int b = 32;
        int array_size = (int)Math.ceil((double)Math.abs(b-a)/32);
        int[] arr = new int[array_size];

        for(int i=a; i<=b; i++){
            if(i%2 == 0 || i%5 == 0){
                setBit(arr, i-a);
            }
        }

        for(int i=a; i<=b; i++){
            if(checkBit(arr, i-a)){
                System.out.print(i + " ");
            }
        }

    }
}
